src/test/kotlin/uk/gov/dluhc/printapi/service/PrintResponseProcessingServiceTest.kt

```diff
@@ -166,6 +174,80 @@ class PrintResponseProcessingServiceTest {
             assertThat(captor.allValues[1]).usingRecursiveComparison().isEqualTo(listOf(expectedPrintDetails2))
             verify(idFactory).requestId()
         }
+
+        @Test
+        fun `should update all certificates within each batch`() {
```

Comment:
Mirrored existing test, but would probably prefer this as separate tests for `failed batch` and `success batch` and possibly `multiple batches`